realm/realm-library/src/androidTest/java/io/realm/rule/RunInLooperThread.java

```diff
@@ -83,13 +82,11 @@ public void run() {
                             backgroundLooper[0] = Looper.myLooper();
                             try {
                                 realm = Realm.getInstance(realmConfiguration);
-
                                 base.evaluate();
-
                                 Looper.loop();
                             } catch (Throwable e) {
                                 threadAssertionError[0] = e;
-
+                                unitTestFailed = true;
```

Comment:
The following cleanup logic requires that we catch the exception and handle it outside the thread it was caught on, so in this case I think it is okay the way it is.